core/query_engine.go

```diff
@@ -96,7 +103,13 @@ func (q *QueryEngine) Sync() error {
 			if _, ok := cradle.deps[Where]; !ok {
 				return lucyErr.QueryDependencyNotSatisfied
 			}
-			cradle.Exps.Push(qr.Params.(string))
+			param := qr.Params.(string)
+
+			if q.Runtime.CheckForInjection(param) {
+				return lucyErr.QueryInjectionDetected
```

Comment:
Error struct should return details like the severity of injection. For this purpose use custom errors rather than simply assigning variables to `errors.New()`.

Reply:
This will give us much more control on the data of errors that we can serve